frontendUiImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -155,7 +156,9 @@ private IGitRepositorySelectionProvider getGitRepositorySelectionProvider() {
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
     GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
-    project.getMessageBus().connect().subscribe(topic, listener);
+    val messageBusConnection = project.getMessageBus().connect();
+    messageBusConnection.subscribe(topic, listener);
+    Disposer.register(project, messageBusConnection);
```

Comment:
There are at least two places where `if (!project.isDisposed())` is checked as a poor man's measure to fix this bug... they can possibly be removed now (at least for the sake of a long-running test if `Disposer.register` calls are enough)

Reply:
It's not like that. The cases touched on here are related to the message bus and the topic subscription. 

The other cases have nothing to do with the message bus so there is nothing to register by Disposer.

Reply:
Oh! okay